src/Util/DotEnvLoader.php

```diff
@@ -0,0 +1,124 @@
+<?php
+namespace SilverStripe\Upgrader\Util;
+
+use M1\Env\Parser;
+use SilverStripe\Upgrader\CodeCollection\CodeChangeSet;
+
+/**
+ * Load an existing `.env` file and provide a way for merging it with a new set of value.
+ *
+ * Loads environment variables from .env files
+ *
+ */
+class DotEnvLoader
+{
+
+    /**
+     * Full path to an env file that is read (if available). That's also where the results will be written.
+     * @var string
+     */
+    private $envFilePath;
+
+    /**
+     * Existing content of the .env file.
+     * @var [type]
+     */
+    private $inputContent = '';
+
+    /**
+     * Content that will be outputted to the .env file.
+     * @var [type]
+     */
+    private $outputContent;
+
+    /**
+     * @param string $envFilePath File Path to `.env` file
+     * @param array  $consts List of variables that should be added to envrionment file.
+     */
+    public function __construct(string $envFilePath, array $consts)
+    {
+        $this->envFilePath = $envFilePath;
+
+        $this->inputContent = $this->readFile();
```

Comment:
Given that the constructor references almost exclusively private methods, think about how someone extending this class could want to modify this. Not so much a problem in this case, but more in the future when we are developing new core classes. :)